src/frontend/js/classes/sidebar-filter.js

```diff
@@ -0,0 +1,341 @@
+/**
+ * HEIGHT of the header in px
+ */
+const HEADER_HEIGHT = 56;
+
+/**
+ * Sidebar Search module.
+ */
+export default class SidebarFilter {
+  /**
+   * CSS classes
+   *
+   * @returns {Record<string, string>}
+   */
+  static get CSS() {
+    return {
+      sectionHidden: 'docs-sidebar__section--hidden',
+      sectionTitle: 'docs-sidebar__section-title',
+      sectionTitleSelected: 'docs-sidebar__section-title--selected',
+      sectionTitleActive: 'docs-sidebar__section-title--active',
+      sectionList: 'docs-sidebar__section-list',
+      sectionListItem: 'docs-sidebar__section-list-item',
+      sectionListItemWrapperHidden: 'docs-sidebar__section-list-item-wrapper--hidden',
+      sectionListItemSlelected: 'docs-sidebar__section-list-item--selected',
+      sidebarSearchWrapperMac: 'docs-sidebar__search-wrapper-mac',
+      sidebarSearchWrapperOther: 'docs-sidebar__search-wrapper-other',
+    };
+  }
+
+  /**
+   * Creates base properties
+   */
+  constructor() {
+    /**
+     * Stores refs to HTML elements needed for sidebar filter to work.
+     */
+    this.sidebar = null;
+    this.sections = [];
+    this.sidebarContent = null;
+    this.search = null;
+    this.searchResults = [];
+    this.selectedSearchResultIndex = null;
+  }
+
+  /**
+   * Initialize sidebar filter.
+   *
+   * @param {HTMLElement[]} sections - Array of sections.
+   * @param {HTMLElement} sidebarContent - Sidebar content.
+   * @param {HTMLElement} search - Search input.
+   * @param {Function} setSectionCollapsed - Function to set section collapsed.
+   */
+  init(sections, sidebarContent, search, setSectionCollapsed) {
+    // Store refs to HTML elements.
+    this.sections = sections;
+    this.sidebarContent = sidebarContent;
+    this.search = search;
+    this.setSectionCollapsed = setSectionCollapsed;
+    let className = SidebarFilter.CSS.sidebarSearchWrapperOther;
+
+    // Search initialize with platform specific shortcut.
+    if (window.navigator.userAgent.indexOf('Mac') != -1) {
+      className = SidebarFilter.CSS.sidebarSearchWrapperMac;
```

Comment:
are you sure we need to change CSS class for that? Maybe just change the textContent of the shortcut element?

Reply:
done